### ISSUE TITLE
Fix missing/inconsistent docblocks

### ISSUE THREAD

**Issue by cleentfaar** [guzzle/guzzle#872]
tl:dr;
Are you open to merging a PR containing fixed doc-blocks for all classes in the library?

Hey @mtdowling / contributors

First of all, thanks a lot for making this library, I've been using it in quite a few of my home- and workprojects now and enjoying it's easy-of-use a lot.

Working with it so often in my IDE (happens to be phpstorm), I have been stumbling on to something that got in my way a little too often; the lack or inconsistent use of phpdoc-blocks in the Guzzle classes. Especially in the public methods of the `Request`, `Response` and `Client` classes, which most users will probably need to access in their code in some way or another.

This makes using IDEs for writing Guzzle-related code less intuitive, as most IDEs rely on proper phpdocs for some of their auto-completion and error-detection strategies. Aside from that, I think these days everyone should doc-block (at least) their public methods and properties, but that may be open to a discussion which I do not mean to get into here :smile:.

On that note I was wondering if you were open to potentially merging a PR containing a big refactoring/addition of phpdocs in all classes of the guzzle library. 

Having all the classes properly documented would really improve my IDE's auto-completion functionality, as well as solve some false-positive error-traces that an IDE may find, such as wrong/missing `@return` statements making them lose track of what a value represents.

As there may be quite some work in figuring out the exact docblocks that are required for each part of the code, I hope any of you will let me know if this is something that is not open to contributions. If it's all good, I'll get a PR ready soon.
